### ISSUE TITLE
Allow injection of custom DataDifferencer

### ISSUE THREAD

**Issue by timkung** [silverstripe/silverstripe-versioned#247]
In SS3 in the `Versioned` class, the `compareVersions` method used to call: `$diff = DataDifferencer::create($fromRecord, $toRecord);` which allowed a custom DataDifferencer to be injected.

This has now changed to: `$diff = new DataDifferencer($fromRecord, $toRecord);`.

There is a case where a custom differencer would be useful to provide the ability to show differences between Page versions and their associated models, e.g. attached content blocks.

**Comment by robbieaverill** [comment]
Fair enough - fancy a PR? I'd suggest `Injector::inst()->create()` instead of `::create()`, otherwise you'll need to add `Injectable` which may be considered a breaking change

**Comment by timkung** [comment]
Update: Have just found you can now use the hooks in `DataObjectVersionFormFactory` to update the fields displayed when viewing the History (when viewing a single version and comparing versions).

Can close this unless you feel this might still be valuable?

**Comment by michalkleiner** [comment]
I think this would still be a valuable PR as it's basically a regression from SS3, so keeping things on par would be great.

**Comment by timkung** [comment]
Great no worries, will create one and send through :)
